components/script/dom/webglrenderingcontext.rs

```diff
@@ -527,8 +530,17 @@ impl WebGLRenderingContext {
         }
 
         // NOTE: width and height are positive or zero due to validate()
-        let expected_byte_length = width * height * element_size * components / components_per_element;
-        return Ok(expected_byte_length);
+        if height == 0 {
+            return Ok(0);
```

Comment:
nit: Since you early-return, there's no need for the else branch, but not a big deal.